src/Hazelcast.Net/Serialization/Compact/ICompactReader.cs

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Hazelcast.Models;
+
+#nullable enable
+
+namespace Hazelcast.Serialization.Compact
+{
+    /// <summary>
+    /// Defines methods for reading fields from a compact-serialized blob.
+    /// </summary>
+    public interface ICompactReader
+    {
```

Comment:
Is `ReadChar `discarded intentionally?

Reply:
"char" is not a valid compact `FieldKind`. It _is_ defined (`FieldKind.Char`) because `FieldKind` could also be used with Portable (somehow) but it is not supported by compact. There is no `readChar` method in Java's `CompactReader` interface.